### PR TITLE
DOC: Remove broken Kepler link

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to remove broken link referencing a PDF that no longer exists. I am letting the linkcheck run here, hence the "Extra CI" label but will cancel irrelevant jobs.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label.
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by pllim** [comment]
@tomdonaldson will provide new link after getting approval from STScI.

**Comment by barentsen** [comment]
Do we expect the URLs to the Kepler archive manual and instrument handbook to change in the near future?

If so, we may also want to look into updating the ADS records for these two documents, i.e.

* ADS bibcode `2016ksci.rept....9T` (Archive Manual) links to `https://archive.stsci.edu/kepler/manuals/archive_manual.pdf`
* ADS bibcode `2016ksci.rept....1V` (Instrument Handbook) links to `https://archive.stsci.edu/kepler/manuals/KSCI-19033-002.pdf`

The URLs to these documents have been used fairly frequently in the literature, e.g. a Google web search for `site:arxiv.org https://archive.stsci.edu/kepler/manuals/archive_manual.pdf` yields 112 results (likely because we never advertised a DOI for these documents).  It is of course unreasonable to expect that a URL will exist forever...

File: docs/timeseries/index.rst [hunk 1]
```diff
@@ -42,8 +42,7 @@ source::
 .. note::
     The light curve provided here is handpicked for example purposes. For
     more information about the Kepler FITS format, see
-    `Section 2.3.1 of the Kepler Archive Manual
-    <https://archive.stsci.edu/kepler/manuals/archive_manual.pdf#page=16>`_.
+    Section 2.3.1 of the Kepler Archive Manual.
```
Comment (tomdonaldson) [inline docs/timeseries/index.rst:45]: The new link is included in the section below.  It is not expected to change in the near term.  There is currently not a DOI for that document.
```suggestion
    `Section 2.3.1 of the Kepler Archive Manual. <https://archive.stsci.edu/files/live/sites/mast/files/home/missions-and-data/k2/_documents/MAST_Kepler_Archive_Manual_2020.pdf>`_.
```

**Comment by pllim** [comment]
@barentsen , Kepler's ADS records are beyond the scope of `astropy`. Perhaps you could contact the group now responsible for such an update.

File: docs/timeseries/index.rst (hunk 1 above)
Comment (pllim) [reply docs/timeseries/index.rst]: Thank you, @tomdonaldson !

**Comment by scfleming** [comment]
This is not part of this pull request or Astropy, but as an FYI to @barentsen, the ADS record pointing to the 2016 version of the KAM has been fixed (it was temporarily broken from some work we did within the past week).  It would be nice to get the modern version of the KAM into ADS, but I don't know how that process worked back in 2016 nor if I'd have bandwidth to pursue that.  The URLs are NOT expected to change from where they are today anytime in the near future.

**Comment by pllim** [comment]
Thanks, everyone! Link check passed and there is no reason to wait a few hours for unrelated cron jobs. Merging now.

**Pull request merged**